client/my-sites/post-type-list/index.jsx

```diff
@@ -153,6 +154,7 @@ class PostTypeList extends Component {
 						siteId={ siteId }
 						query={ { ...query, page } } />
 				) ) }
+				{ siteId && <QueryPostTypes siteId={ siteId } /> }
```

Comment:
Alternatively, while it goes against our "keeping fresh" methodology of query component mounting, we could render the `<QueryPostTypes />` within the menu items, but only if types aren't yet known for the site.